actions/lib/base.py

```diff
@@ -0,0 +1,87 @@
+import json
+import requests
+import urllib3
+
+from st2common.runners.base_action import Action
+
+
+class SplunkBaseAction(Action):
+    """Base functinos for calling other api actinos."""
+    def __init__(self, config):
+        super(SplunkBaseAction, self).__init__(config)
+        # Validate config is set
+        if config is None:
+            raise ValueError("No Splunk configuration details found")
+        if "splunk_instances" in config:
+            if config['splunk_instances'] is None:
+                raise ValueError("'splunk_instances' config empty.")
+            else:
+                pass
+        else:
+            raise ValueError("No Splunk configuration details found")
+
+    def instance_details(self, instance):
+        """Collect instance details from config."""
+        # requests.packages.urllib3.disable_warnings()
+        urllib3.disable_warnings(urllib3.exceptions.InsecureRequestWarning)
+
+        # Build the base splunk url
+        splunk_config = self.config['splunk_instances'].get(instance)
+
+        host = splunk_config.get('host')
+        port = splunk_config.get('port')
+        scheme = splunk_config.get('scheme')
+        verify = splunk_config.get('verify')
+
+        try:
+            if splunk_config.get('splunkToken'):
+                splunk_token = splunk_config.get('splunkToken')
+            else:
+                username = splunk_config.get('username')
+                password = splunk_config.get('password')
+                splunk_token = None
+        except BaseException as err:
+            raise Exception(
+                "Failed to connect to Splunk"
+                "Instance {} with error {}".format(splunk_config, err))
+
+        base_url = "{}://{}:{}/".format(scheme, host, port)
+
```

Comment:
I'm not sure what the main driver was for the refactor, but up until line 46 in this new base action is the same code as performed in the search.py. Not essential, but it might be worth considering having this first part in its own method, so that search.py could inherit from this base, either in this PR or a future one.